pkg/admission/machines.go

```diff
@@ -39,13 +44,18 @@ func (ad *admissionData) mutateMachines(ar admissionv1beta1.AdmissionReview) (*a
 		if oldMachine.Spec.Name != machine.Spec.Name && machine.Spec.Name == machine.Name {
 			oldMachine.Spec.Name = machine.Spec.Name
 		}
-		// Allow mutation when oldMachine has Initializers on it
-		if oldMachine.Initializers == nil || len(oldMachine.Initializers.Pending) == 0 {
+		// Allow mutation when:
+		// * oldMachine has Initializers on it
+		// * machine has the `MigrationBypassSpecNoModificationRequirementAnnotation` annotation (used for type migration)
+		_, hasBypassSpecNoModificationRequirementAnnotation := machine.Annotations[BypassSpecNoModificationRequirementAnnotation]
```

Comment:
Not sure if this is the most elegant approach. Definitely open for ideas here

Reply:
I would prefer checking the value to be more precise:
```
bypassValidationForMigration := machine.Annotations[BypassSpecNoModificationRequirementAnnotation] == "true"
```

But no hard opinion here - i dont think the `exists-in-map` approach is bad